lib/src/input/figma/entities/style/figma_fill.dart

```diff
@@ -1,26 +1,78 @@
 import 'package:json_annotation/json_annotation.dart';
+import 'package:pbdl/src/input/figma/entities/style/fill/figma_fill_factory.dart';
 import 'package:pbdl/src/pbdl/pbdl_fill.dart';
-
+import '../../../../../pbdl.dart';
 import 'figma_color.dart';
+
 part 'figma_fill.g.dart';
 
 @JsonSerializable()
 class FigmaFill {
   FigmaColor color;
 
-  FigmaFill(this.color, [this.isEnabled = true]);
+  @JsonKey(defaultValue: 100)
+  num opacity;
+
+  String blendMode;
+
+  String type;
 
   @JsonKey(defaultValue: true)
-  bool isEnabled;
+  bool visible;
 
-  Map<String, dynamic> toJson() => _$FigmaFillToJson(this);
+  FigmaFill(
+    this.opacity,
+    this.blendMode,
+    this.type,
+    this.visible,
+    this.color,
+  );
+
+  Map<String, dynamic> toJson() {}
+  FigmaFill createFigmaFill(Map<String, dynamic> json) =>
+      FigmaFill.fromJson(json);
   factory FigmaFill.fromJson(Map<String, dynamic> json) =>
-      _$FigmaFillFromJson(json);
+      FigmaFillFactory.getFill(json);
 
   PBDLFill interpretFill() {
     return PBDLFill(
-      color?.interpretColor(),
-      isEnabled,
+      opacity: opacity,
+      blendMode: blendMode,
+      type: type,
+      isEnabled: visible,
+      color: color.interpretColor(),
     );
   }
+
```

Comment:
I think we should try to remove all commented code if possible in a future refactor, not high priority to remove this at the moment though 👍🏾